src/inspector/view/qml/MuseScore/Inspector/notation/barlines/BarlineSettings.qml

```diff
@@ -56,19 +56,28 @@ Column {
         navigationPanel: root.navigationPanel
         navigationRowStart: root.navigationRowStart
 
-        model: [
-            { text: qsTrc("symUserNames", "Single barline"), value: BarlineTypes.TYPE_NORMAL },
-            { text: qsTrc("symUserNames", "Double barline"), value: BarlineTypes.TYPE_DOUBLE },
-            { text: qsTrc("symUserNames", "Left (start) repeat sign"), value: BarlineTypes.TYPE_START_REPEAT },
-            { text: qsTrc("symUserNames", "Right (end) repeat sign"), value: BarlineTypes.TYPE_END_REPEAT },
-            { text: qsTrc("symUserNames", "Right and left repeat sign"), value: BarlineTypes.TYPE_END_START_REPEAT },
-            { text: qsTrc("symUserNames", "Dashed barline"), value: BarlineTypes.TYPE_DASHED },
-            { text: qsTrc("symUserNames", "Final barline"), value: BarlineTypes.TYPE_FINAL },
-            { text: qsTrc("symUserNames", "Dotted barline"), value: BarlineTypes.TYPE_DOTTED },
-            { text: qsTrc("symUserNames", "Reverse final barline"), value: BarlineTypes.TYPE_REVERSE_END },
-            { text: qsTrc("symUserNames", "Heavy barline"), value: BarlineTypes.TYPE_HEAVY },
-            { text: qsTrc("symUserNames", "Heavy double barline"), value: BarlineTypes.TYPE_DOUBLE_HEAVY },
-        ]
+        visible: root.barlineSettingsModel && root.barlineSettingsModel.isStyleChangingAllowed
+
+        function getModel() {
+            let model = [
+                        { text: qsTrc("symUserNames", "Single barline"), value: BarlineTypes.TYPE_NORMAL },
+                        { text: qsTrc("symUserNames", "Double barline"), value: BarlineTypes.TYPE_DOUBLE },
+                        { text: qsTrc("symUserNames", "Left (start) repeat sign"), value: BarlineTypes.TYPE_START_REPEAT },
+                        { text: qsTrc("symUserNames", "Right (end) repeat sign"), value: BarlineTypes.TYPE_END_REPEAT },
+                        { text: qsTrc("symUserNames", "Right and left repeat sign"), value: BarlineTypes.TYPE_END_START_REPEAT },
+                        { text: qsTrc("symUserNames", "Dashed barline"), value: BarlineTypes.TYPE_DASHED },
+                        { text: qsTrc("symUserNames", "Final barline"), value: BarlineTypes.TYPE_FINAL },
+                        { text: qsTrc("symUserNames", "Dotted barline"), value: BarlineTypes.TYPE_DOTTED },
+                        { text: qsTrc("symUserNames", "Reverse final barline"), value: BarlineTypes.TYPE_REVERSE_END },
+                        { text: qsTrc("symUserNames", "Heavy barline"), value: BarlineTypes.TYPE_HEAVY },
+                        { text: qsTrc("symUserNames", "Heavy double barline"), value: BarlineTypes.TYPE_DOUBLE_HEAVY },
+                    ]
+            return model.filter(function(item) { 
+                return (item.value != BarlineTypes.TYPE_START_REPEAT && item.value != BarlineTypes.TYPE_END_START_REPEAT) ||
```

Comment:
This code should be in the model. qml isn't the right place for such logic

Reply:
Basically I'm suggesting to leave the list of barline types defined in the qml file where it is now, but call a filter function that's exposed by the C++ model to determine which ones should be shown.  Does that sound reasonable?

Reply:
Or should I use SortFilterProxyModel? There's quite a few examples of that, though I'm not sure about one using a Q_INVOKABLE filter.

Reply:
SortFilterProxyModel seemed excessively complex for what I needed, so I hope my "compromise" solution is ok...even though I admit it's the only use of the javascript filter function in any of the qml models.